src/slices/chat-slice.ts

```diff
@@ -374,14 +441,23 @@ export const chatSlice = createSlice({
       state.chatId = action.payload;
     },
     addMessage: (state, action: PayloadAction<Message>) => {
-      state.messages = filterDuplicatMessages([...state.messages, action.payload]);
+      state.messages = filterDuplicatMessages([
+        ...state.messages,
+        action.payload,
+      ]);
+    },
+    addMessageToTop: (state, action: PayloadAction<Message>) => {
+      state.messages = [action.payload, ...state.messages];
     },
```

Comment:
The only functional change in this file, all others are formatting changes.

Since there are a lot of formatting issues in this repo, we need to think how to deal with them. Fixing these as we work on individual task PRs makes reviewing hard. What I did once in another project was to create one PR where all formatting was fixed in one go — and then also add a CI formatting check. I will think a bit more about this.